source/WeightedList.h

```diff
@@ -86,4 +92,14 @@ const Type &WeightedList<Type>::Get() const
 
 
 
+template<class Type>
+void WeightedList<Type>::RecalculateWeight()
+{
+	total = 0;
+	for(const auto &choice : choices)
+		total += choice.Weight();
```

Comment:
Since this function is only used when erasing, you can simplify the implementation by reducing `total` by the weight of the element to remove. That way you avoid iterating through the whole list.